src/models/StyledComponent.js

```diff
@@ -275,7 +280,12 @@ export default function createStyledComponent(target: Target, options: Object, r
    * instead of extending ParentComponent to create _another_ interim class
    */
   const WrappedStyledComponent = React.forwardRef((props, ref) => (
-    <ParentComponent {...props} forwardedClass={WrappedStyledComponent} forwardedRef={ref} />
+    <ParentComponent
+      {...props}
+      forwardedClass={WrappedStyledComponent}
+      forwardedRef={ref !== null ? ref : props.forwardedRef}
```

Comment:
Thinking about this some more, I believe we can just start implementing an early pass at https://github.com/styled-components/styled-components/pull/2093 by making `forwardedRef` -> `$forwardedRef` and omitting it during prop computation. That way no new prop has to be added and it's just an implementation detail.

What do you think?

Reply:
Ah I see. I like the idea of "transient props" better than this manual props drilling logic. This PR is a more clever (not in the good way) solution.

Reply:
I'm gonna go ahead and close this. See if I can't help out with the other effort instead. Thanks!